### ISSUE TITLE
Audio/Video not working through foundry

### ISSUE THREAD

**Issue by NeoEvaX** [luvolondon/fvtt-module-jitsiwebrtc#34]
To add to previous bug posted.

When using module through foundry each user seems to randomly have audio or video working. It seems like a semi random who has what working.

When "Allow standalone Jitsi users" is selected, and users connect to outside jitsi server directly, they are able to chat without difficulty to others doing the same thing, but are not able to see the video of users still connected through foundry. 

Seeing the stand alone Jitsi server working for all users, it makes me think the issue resides somewhere within Foundry or the Module itself. I do feel like it has gotten a bit worse with last update, but similar issues existed before Module version 0.4 and Foundry 0.7.5.

Foundry -> Foundry = intermittent issues/Random Video/Audio 
Foundry -> Stand alone Jitsi = Foundry people can see Jitsi people, but Jitsi people can't see (but can hear) Foundry
Jitsi -> Jitsi = Everything works great.

Using:
Foundry 0.7.5
Jitsi Module: 0.4
Foundry installed on local machine via Docker (with SSL and open ports etc)
also Foundry installed on Digital Ocean via Docker following guide located here: [https://www.youtube.com/watch?v=1Mr5sWkPXSk](url)
Jitsi installed on Digital Ocean
Jitsi updated fully on server

Users all using up to date Chrome. No obvious console lines I can see.

**Comment by feltz916** [comment]
following up, that I have similar connectivity issues. I have a game with 7 players, and had a tough time getting everyones a/v working. Out of the 7, it took about 30 minutes to get everyone working, and there was one player who I could never get their audio working. So he ended up using text only.

The good:
Once someone's audio and video was working, it seemed stable and stayed that way. Did not have disconnects or anything like that.

The bad:
Very hard to troubleshoot players connectivity issues, since we couldnt chat.
It seemed as if there was no way a player could initialize and/or change their hardware settings, without disconnecting and reconnecting. Changing audio input sources would not work while in the session, they had to set the device, and reconnect for it to initialize.

Foundry 0.7.5 (Local Host)
Jitsi Module: 0.4
Jitsi Server - Digital Ocean

**Comment by RDPrint** [comment]
I had the Jitsi module working fine on 0.6.5 but i upgraded to 0.7.5 and found similar issues to the above.

Issue:
Players were connecting with no audio or video coming through. blanking their camera and causing them to reload would occasionally bring them back but some players couldn't see others that I could, others couldn't hear each other. 
Tried the standard WebRTC in Foundry with the same result so it could be a deeper issue at play

Using the Jitsi module in the default config with no traditional paid hosting

**Comment by therpgdaily** [comment]
I'll add another 2 cents onto this. 

Foundry 0.7.5 (Digital Ocean)
Jitsi Module 0.4
Jitsi Server (Digital Ocean - Separate droplet from the foundry server)
default setting on the jitsi server.

My main problem was that when my audio was set to voice activation it would not work when the module was active. I did find the workaround that setting my broadcast mode to always enabled seemed to set it to voice activated.  I've reproduced this effect with and without other modules active in foundry.

But the video connectivity then became an issue, I ran tests using chrome for broadcasting and opened a firefox window for monitoring. the audio would come through but the video would sit on a freeze frame in firefox.
I tried it again using a chrome incognito tab to monitor it and the audio came through but no video.

When I use the jitsi server on its own it works perfectly fine.

**Comment by Ektorx** [comment]
so same issue here 

Foundry 0.7.5 (Digital Ocean)
Jitsi Module 0.4
Jitsi Server (Digital Ocean - Separate droplet from the foundry server)
default setting on the jitsi server.

We will have, randomly, either audio only, audio and video, or nada, for each player.
if i do change sound volume, i need to refresh my page to have it really changed
a weird issue : i do use a dedicated jitsi server, and do connect through chrome to it
if i do refresh my local chrome, or change something like activating/deactivating an audio or video of one the jistsi stream on my client, it will affect everyone.
Example :
we were 6 players, few were working well, few were only audio, and some nothing at all.
I cut audio and video of one not working, then reactivate it.
every streams will be refreshed, and randomly some will work, some partially. This will affect every clients.

at the end, it s like each time throw a d6 for each client, hoping doing a 4+ to make it work for each one.

plus so the issue with taking in account any change, looks like we need to refresh the page to have any tweak validated, like changing sound volume, or changing an option of the voice activation. I had several times issues like : i am on "hit a key" activation, but in reality it was on voice activation, so everyone was hearing me without any control on my side.

**Comment by bekriebel** [comment]
I have just updated the lib-jitsi-meet library using the latest version from beta.meet.jit.si. In my quick testing, this seems to help with some of the connection issues. Can folks please test with the new 0.4.1 version and let me know if it is still acting up?

Note: I do still expect there to be a few oddities around changing settings, but I'm hoping the main connectivity issues will be resolved.

**Comment by Mist-Hunter** [comment]
I seem to be having the same problem described above, with 0.4.1. Voice-activation no longer works (as with 0.4.0). I'll try and and provide more info soon.

**Comment by bekriebel** [comment]
I can check on voice activation, but I would suggest just using "always on". Jitsi already does audio-leveling and will only transmit audio when you cross a threshold, just like Voice Activation. I think the two systems actually step on each other and I may just remove the Voice Activation option from the module since Jitsi is already doing this natively.

@Mist-Hunter, are you still facing the other issues, or just Voice Activation?

**Comment by Ektorx** [comment]
ah voice activation refused to work on my side too.
but yes prefer always activated or key activation, voice activation does eat some words
else worked fine with 6 players

**Comment by leWebslinger** [comment]
Hi, I am experiencing all the above issues. 
running ubuntu locally
Foundry(0.7.5)
JitsiWebrtc(0.4.1)
my subdomain where I can use the indipendent jitsi client is perfectly working.
The connections done using the paraller link (given in the settings of the jitsi module) are stable and functioning for everyone but doing this for every partecipant can be confusing and time consuming, plus you lose the colour assignment for players.

**Comment by leWebslinger** [comment]
After the session of yesterday I haven't experienced any major issue.
Still when people muted themselves for temporary privacy their audio input would not return when they were unmuting themselves.

**Comment by luvolondon** [comment]
> Issue:
> Players were connecting with no audio or video coming through. blanking their camera and causing them to reload would occasionally bring them back but some players couldn't see others that I could, others couldn't hear each other.
> Tried the standard WebRTC in Foundry with the same result so it could be a deeper issue at play
> 
> Using the Jitsi module in the default config with no traditional paid hosting

I think using the public Jitsi server (beta.meet.jit.si) atm is no stable solution for game play. I´ve tried to get this working the last several days without success. This was different when all this started, the beta servers where quite good and reliable. But now using the lib-jitsi-meet lib this no longer works well. There must be some secret sauce missing to use their servers from the API. I highly suggest to run your own server.

**Comment by luvolondon** [comment]
> 
> 
> following up, that I have similar connectivity issues. I have a game with 7 players, and had a tough time getting everyones a/v working. Out of the 7, it took about 30 minutes to get everyone working, and there was one player who I could never get their audio working. So he ended up using text only.
> 
> The good:
> Once someone's audio and video was working, it seemed stable and stayed that way. Did not have disconnects or anything like that.
> 
> The bad:
> Very hard to troubleshoot players connectivity issues, since we couldnt chat.
> It seemed as if there was no way a player could initialize and/or change their hardware settings, without disconnecting and reconnecting. Changing audio input sources would not work while in the session, they had to set the device, and reconnect for it to initialize.
> 
> Foundry 0.7.5 (Local Host)
> Jitsi Module: 0.4
> Jitsi Server - Digital Ocean

Getting 7 players setup for a A/V session can definitely take some time. I´ld have a discord channel running in parallel to have a way to chat. What problems did you experience and solve in what way?

**Comment by NeoEvaX** [comment]
Finally got around to playing another session. Things went better, but we still had issues and had to bounce around settings.

One more error that might help. At one point I turned the module back on, updated, and connected to Foundry VTT and got this error in console every few seconds:
`[modules/RTC/BridgeChannel.js] <l._send>:  Bridge Channel send: no opened channel.`

Changing to my custom server worked for awhile too, but we started to have other issues. Users audio freezing etc. Though not quite as bad as before.

So what we ended up doing was switching between not using module and using built in VA system, using module with Foundry VTT, then module with custom server.. It ended up working enough, but still odd.

I did notice that users would occasionally get super blown out audio. But it might only affect 1 other user.. So User A,B,C. Everyone hears eachother fine.. then B starts to hear C super blown out. B cant turn down the volume of C.. once C refreshes it goes away. But that whole time A is fine. Refreshing seemed to help the issue though..
